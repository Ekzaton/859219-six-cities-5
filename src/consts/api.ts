export const BASE_URL = `https://6.react.pages.academy/six-cities`;
export const REQUEST_TIMEOUT = 5000;
