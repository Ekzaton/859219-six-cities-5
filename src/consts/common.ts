export enum AppRoute {
  MAIN = `/`,
  LOGIN = `/login/`,
  FAVORITES = `/favorites/`,
  OFFER_ID = `/offer/:id/`,
  OFFER = `/offer/`
}

export enum AuthStatus {
  AUTH = `auth`,
  NO_AUTH = `no-auth`
}

export enum FilteringType {
  PARIS = `Paris`,
  COLOGNE = `Cologne`,
  BRUSSELS = `Brussels`,
  AMSTERDAM = `Amsterdam`,
  HAMBURG = `Hamburg`,
  DUSSELDORF = `Dusseldorf`
}

export enum SortingType {
  POPULAR = `Popular`,
  PRICE_LOW_TO_HIGH = `Price: low to high`,
  PRICE_HIGH_TO_LOW = `Price: high to low`,
  TOP_RATED_FIRST = `Top rated first`
}
