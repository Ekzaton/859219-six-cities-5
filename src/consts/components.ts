export const OFFER_IMAGES_COUNT = 6;
export const OFFER_REVIEWS_COUNT = 10;
export const RATING_STARS_COUNT = 5;
export const REG_EXP_EMAIL = /^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/;
export const REG_EXP_PASSWORD = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/;
export const DELAY_DURATION = 250;

export enum BtnBigSize {
  WIDTH = 31,
  HEIGHT = 33
}

export enum BtnSize {
  WIDTH = 18,
  HEIGHT = 19
}

export enum BtnType {
  PLACE_CARD = `place-card`,
  PROPERTY = `property`
}

export enum CardImgSize {
  WIDTH = 260,
  HEIGHT = 200
}

export enum CardType {
  CITIES = `cities`,
  FAVORITES = `favorites`,
  NEAR_PLACES = `near-places`
}

export enum FavCardImgSize {
  WIDTH = 150,
  HEIGHT = 110
}

export enum HeaderType {
  MAIN = `main`,
  LOGIN = `login`
}

export enum MapIconSize {
  WIDTH = 30,
  HEIGHT = WIDTH
}

export enum MapIconUrl {
  PIN = `img/pin.svg`,
  PIN_ACTIVE = `img/pin-active.svg`
}

export enum MapLayer {
  TEMPLATE = `https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`,
  ATTRIBUTION = `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
}

export enum MapType {
  CITIES = `cities`,
  PROPERTY = `property`
}

export enum ReviewDate {
  LOCALE = `en-US`,
  MONTH = `long`,
  YEAR = `numeric`
}

export enum ReviewLehgth {
  MIN = 50,
  MAX = 300
}
