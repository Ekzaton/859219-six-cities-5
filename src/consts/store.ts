export enum ActionType {
  CHANGE_ACTIVE_OFFER_ID = `CHANGE_ACTIVE_OFFER_ID`,
  CHANGE_CURRENT_CITY = `CHANGE_CURRENT_CITY`,
  CHANGE_CURRENT_SORTING = `CHANGE_CURRENT_SORTING`,
  CHANGE_OFFER_FAVORITE_STATUS = `CHANGE_OFFER_FAVORITE_STATUS`,
  GET_ALL_OFFERS = `GET_ALL_OFFERS`,
  GET_FAVORITE_OFFERS = `GET_FAVORITE_OFFERS`,
  GET_NEARBY_OFFERS = `GET_NEARBY_OFFERS`,
  GET_SINGLE_OFFER = `GET_SINGLE_OFFER`,
  GET_REVIEWS = `GET_REVIEWS`,
  GET_USER = `GET_USER`,
  REDIRECT_TO_ROUTE = `REDIRECT_TO_ROUTE`,
  SET_AUTH_STATUS = `SET_AUTH_STATUS`,
  SET_INITIAL_STATE = `SET_INITIAL_STATE`,
  SET_IS_ALL_OFFERS_LOADING = `SET_IS_ALL_OFFERS_LOADING`,
  SET_IS_ALL_OFFERS_LOADING_ERROR = `SET_IS_ALL_OFFERS_LOADING_ERROR`,
  SET_IS_FAVORITE_OFFERS_LOADING = `SET_IS_FAVORITE_OFFERS_LOADING`,
  SET_IS_FAVORITE_OFFERS_LOADING_ERROR = `SET_IS_FAVORITE_OFFERS_LOADING_ERROR`,
  SET_IS_FAVORITE_SENDING = `SET_IS_FAVORITE_SENDING`,
  SET_IS_FAVORITE_SENDING_ERROR = `SET_IS_FAVORITE_SENDING_ERROR`,
  SET_IS_LOGIN_SENDING = `SET_IS_LOGIN_SENDING`,
  SET_IS_LOGIN_SENDING_ERROR = `SET_IS_LOGIN_SENDING_ERROR`,
  SET_IS_REVIEW_SENDING = `SET_IS_REVIEW_SENDING`,
  SET_IS_REVIEW_SENDING_ERROR = `SET_IS_REVIEW_SENDING_ERROR`,
  SET_IS_SINGLE_OFFER_LOADING = `SET_IS_SINGLE_OFFER_LOADING`,
  SET_IS_SINGLE_OFFER_LOADING_ERROR = `SET_IS_SINGLE_OFFER_LOADING_ERROR`
}

export enum APIEndpoint {
  HOTELS = `/hotels/`,
  NEARBY = `/nearby/`,
  FAVORITE = `/favorite/`,
  COMMENTS = `/comments/`,
  LOGIN = `/login/`,
  LOGOUT = `/logout/`
}
