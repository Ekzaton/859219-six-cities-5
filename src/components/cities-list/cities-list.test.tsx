import React from 'react';
import {Provider} from 'react-redux';
import createMockStore from 'redux-mock-store';
import {render} from '@testing-library/react';

import {CityName} from "../../consts/common";

import CitiesList from "./cities-list";

describe(`Cities list renders correctly`, () => {
  const mockStore = createMockStore([]);
  const store = mockStore();

  it(`for initial city`, () => {
    const {container} = render(
        <Provider store={store}>
          <CitiesList
            currentCity={CityName.PARIS}
          />
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`for selected city`, () => {
    const {container} = render(
        <Provider store={store}>
          <CitiesList
            currentCity={CityName.AMSTERDAM}
          />
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
