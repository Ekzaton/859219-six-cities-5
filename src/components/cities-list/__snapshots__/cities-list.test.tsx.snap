// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cities list renders correctly for initial city 1`] = `
<div>
  <ul
    class="locations__list tabs__list"
  >
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item tabs__item--active"
      >
        <span>
          Paris
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item false"
      >
        <span>
          Cologne
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item false"
      >
        <span>
          Brussels
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item false"
      >
        <span>
          Amsterdam
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item false"
      >
        <span>
          Hamburg
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item false"
      >
        <span>
          Dusseldorf
        </span>
      </a>
    </li>
  </ul>
</div>
`;

exports[`Cities list renders correctly for selected city 1`] = `
<div>
  <ul
    class="locations__list tabs__list"
  >
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item false"
      >
        <span>
          Paris
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item false"
      >
        <span>
          Cologne
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item false"
      >
        <span>
          Brussels
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item tabs__item--active"
      >
        <span>
          Amsterdam
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item false"
      >
        <span>
          Hamburg
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item false"
      >
        <span>
          Dusseldorf
        </span>
      </a>
    </li>
  </ul>
</div>
`;
